### PR TITLE
Adapt TotalVI to new setup

### PR DESCRIPTION
**Pull request by jjhong922**
Adapts TotalVI to the new setup scheme. Required a lot of changes in places that used hardcoded keys or referenced metadata previously stored on the AnnData.

### CODE REVIEW

File: scvi/data/anndata/fields/_obsm_field.py [hunk 1]
```diff
@@ -33,6 +36,104 @@ def attr_name(self) -> str:
...
+                f"Using column names from columns of adata.obsm['{self.attr_key}']"
+            )
+            column_names = list(obsm_data.columns)
+        elif self.colnames_uns_key is not None:
+            logger.info(
+                f"Using protein names from adata.uns['{self.colnames_uns_key}']"
+            )
+            column_names = adata.uns[self.colnames_uns_key]
+        else:
+            logger.info("Generating sequential protein names")
```
Comment (adamgayoso) [inline scvi/data/anndata/fields/_obsm_field.py:107]: should be generic

File: scvi/data/anndata/fields/_obsm_field.py [hunk 2]
```diff
@@ -33,6 +36,104 @@ def attr_name(self) -> str:
...
+    def _setup_column_names(self, adata: AnnData) -> Union[list, np.ndarray]:
+        obsm_data = self.get_field(adata)
+        if self.colnames_uns_key is None and isinstance(obsm_data, pd.DataFrame):
+            logger.info(
+                f"Using column names from columns of adata.obsm['{self.attr_key}']"
+            )
+            column_names = list(obsm_data.columns)
+        elif self.colnames_uns_key is not None:
+            logger.info(
+                f"Using protein names from adata.uns['{self.colnames_uns_key}']"
```
Comment (adamgayoso) [inline scvi/data/anndata/fields/_obsm_field.py:103]: should be generic

File: scvi/data/anndata/fields/_obsm_field.py [hunk 3]
```diff
@@ -33,6 +36,104 @@ def attr_name(self) -> str:
...
+        self, state_registry: dict, adata_target: AnnData, **kwargs
+    ) -> dict:
+        super().transfer_field(state_registry, adata_target, **kwargs)
+        self.validate_field(adata_target)
+        source_n_cols = len(state_registry[self.COLUMN_NAMES_KEY])
+        target_n_cols = self.get_field(adata_target).shape[1]
+        assert source_n_cols == target_n_cols, (
+            f"Target adata.obsm['{self.attr_key}'] has {target_n_cols} which does not match "
+            "the source adata.obsm['{self.attr_key}'] column count of {source_n_cols}."
+        )
```
Comment (adamgayoso) [inline scvi/data/anndata/fields/_obsm_field.py:128]: should we actually check if the column names are the same?

**Review by adamgayoso — CHANGES_REQUESTED** [review]
need to do another review, but looks good

File: scvi/data/anndata/fields/_obsm_field.py [hunk 4]
```diff
@@ -33,6 +36,111 @@ def attr_name(self) -> str:
         return self._attr_name
 
 
+class ObsmField(BaseObsmField):
+    """
+    An AnnDataField for an .obsm field in the AnnData data structure.
```
[the remark this answers is not in the document]
Comment (jjhong922) [reply scvi/data/anndata/fields/_obsm_field.py:41]: AnnDataField doesn't exist only the concept of one (child of a BaseAnnDataField). Do you think I should just link this? Or perhaps ``Type[~scvi...AnnDataField]``.

[the remark this answers is not in the document]
Comment (jjhong922) [reply scvi/data/anndata/fields/_obsm_field.py:41]: Leaving this for now, can change in a later PR.

**Pull request merged**